### PR TITLE
Newdevice: Moxa EtherDevice Industrial Switch

### PR DESCRIPTION
**Pull request by aldemira**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Added support Moxa EtherDevice industrial switches. Detects cpu, memory and sets the right icon.

### CODE REVIEW

File: includes/definitions/moxa-etherdevice.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+os: moxa-etherdevice
```
Comment (laf) [inline includes/definitions/moxa-etherdevice.yaml]: Is that the best OS name?

File: includes/discovery/mempools/moxa-etherdevice.inc.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Copyright (c) 2017 Aldemir Akpinar <aldemir.akpinar@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
```
Comment (laf) [inline includes/discovery/mempools/moxa-etherdevice.inc.php]: Can you clear the blank spaces here pls?

File: includes/discovery/mempools/moxa-etherdevice.inc.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+if ($device['os'] == 'moxa-etherdevice') {
+    echo 'Moxa EtherDevice';
+
+    $total = str_replace('"', "", snmp_get($device, "MOXA-IKS6726A-MIB::totalMemory.0", '-OvQ'));
```
Comment (laf) [inline includes/discovery/mempools/moxa-etherdevice.inc.php]: If you move the mib into mibs/moxa then you can replace the snmp queries with:

`snmp_get($device, "totalMemory.0", '-OvQ', 'MOXA-IKS6726A-MIB')`

File: includes/definitions/moxa-etherdevice.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+icon: moxa
+text: Moxa
+type: switch
+over:
+    - { graph: device_bits, text: 'Device Traffic' }
+    - { graph: device_processor, text: 'CPU Usage' }
+    - { graph: device_mempool, text: 'Memory Usage' }
+discovery:
+    - sysObjectId:
+        - .1.3.6.1.4.1.8691.7.116
```
Comment (laf) [inline includes/definitions/moxa-etherdevice.yaml]: With moving the mib add:

```
mib_dir:
    - moxa
```

File: includes/discovery/processors/moxa-etherdevice.inc.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+
+if ($device['os'] == 'moxa-etherdevice') {
+    echo 'Moxa EtherDevice Switch : ';
+    $descr = 'Processor';
+    $usage = snmp_get($device, 'cpuLoading30s.0', '-Ovq', 'MOXA-IKS6726A-MIB');
+    echo $usage."\n";
```
Comment (laf) [inline includes/discovery/processors/moxa-etherdevice.inc.php]: Should be `d_echo($usage);`

File: includes/discovery/processors/moxa-etherdevice.inc.php [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    if (is_numeric($usage)) {
+        discover_processor($valid['processor'], $device, 'MOXA-IKS6726A-MIB::cpuLoading30s.0', '0', 'moxa-etherdevice', $descr, '1', $usage, null, null);
```
Comment (laf) [inline includes/discovery/processors/moxa-etherdevice.inc.php]: You should be able to drop `MOXA-IKS6726A-MIB::` here.

File: includes/polling/mempools/moxa-etherdevice-mem.inc.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        freeMemory OBJECT-TYPE
+            "Total size of free dynamic memory"
+     usedMemory OBJECT-TYPE
+            "Total size of used dynamic memory"
+     memoryUsage OBJECT-TYPE
+            "The usage of memory size in %."
+*/
+
+if ($device['os'] == 'moxa-etherdevice') {
+    $perc     = snmp_get($device, "MOXA-IKS6726A-MIB::memoryUsage.0", '-OvQ');
```
Comment (laf) [inline includes/polling/mempools/moxa-etherdevice-mem.inc.php]: I don't think this is used so can be dropped.

File: includes/polling/processors/moxa-etherdevice-cpu.inc.php [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+
+echo 'Moxa EtherDevice CPU Usage';
+
+if ($device['os'] == 'moxa-etherdevice') {
+    $usage = str_replace('"', "", snmp_get($device, 'MOXA-IKS6726A-MIB::cpuLoading30s.0', '-OvQ'));
+    //$proc = $usage;
```
Comment (laf) [inline includes/polling/processors/moxa-etherdevice-cpu.inc.php]: May as well ditch that line.

File: includes/definitions/moxa-etherdevice.yaml (hunk 1 above)
Comment (aldemira) [inline includes/definitions/moxa-etherdevice.yaml]: suggestions welcome :)

**Comment by laf** [comment]
Sorry, should have also said you need to add unit tests here as well :)

**Comment by aldemira** [comment]
Yup, I realised that after the creating the request :) Anyways, I've pushed the tests too.

File: includes/definitions/moxa-etherdevice.yaml (hunk 1 above)
Comment (laf) [inline includes/definitions/moxa-etherdevice.yaml]: I don't have any :)

File: includes/polling/os/moxa-etherdevice.inc.php [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright (c) 2017 Aldemir Akpinar <aldemir.akpinar@gmail.com>
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+
+$version = snmp_get($device, "MOXA-IKS6726A-MIB::firmwareVersion.0", "-OQvs", "");
```
Comment (laf) [inline includes/polling/os/moxa-etherdevice.inc.php]: Need to shift this MOXA-IK....

**Comment by laf** [comment]
Just one last MIB:: move and it's done.

**Comment by laf** [comment]
Actually did spot at the end, no logo :(

**Pull request merged**